[Listbox, OptionList] Right align check icons on selected list items

kyledurand opened the pull request:
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-internal/issues/1503

### WHAT is this pull request doing?

Change CheckIcon placement from right to left alignment. 
Adds a placeholder space for non-selected options

[Figma](https://www.figma.com/file/I3df2veCGRVUuxzpXVE71m/Pickers?type=design&node-id=1655%3A23178&mode=design&t=ojAxTxPUmN0SoZYW-1)

<!--
  Summary of the changes committed.

  Before / after screenshots are appreciated for UI changes. Make sure to include alt text that describes the screenshot.

  Include a video if your changes include interactive content.

  If you include an animated gif showing your change, wrapping it in a details tag is recommended. Gifs usually autoplay, which can cause accessibility issues for people reviewing your PR:

  <details>
    <summary>Summary of your gif(s)</summary>
    <img src="..." alt="Description of what the gif shows">
  </details>
-->

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#install-dependencies-and-build-workspaces)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [ ] Tested a [snapshot](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#-snapshot-releases)
- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

kyledurand committed:
chore: update listbox layout

kyledurand committed:
chore: update Optionlist layout

kyledurand committed:
changeset

kyledurand committed:
fix up story

sophschneider reviewed (approved):
Code and changed stories LGTM!

Also not totally related to your changes but I noticed that the option list item could use a little more padding at the end when not selected, its causing the [popover to grow](https://5d559397bae39100201eedc1-tsymzrugoz.chromatic.com/?path=/story/all-components-optionlist--in-a-popover) when selected right now.

<img width="155" alt="Screenshot 2024-03-07 at 9 42 38 AM" src="https://github.com/Shopify/polaris/assets/20652326/a4a8f175-52a0-47a7-a36c-e7884689b4e7">

kyledurand committed:
update gap

kyledurand committed:
[OptionList] Fix layout shift within popovers (#11736)

Fixes https://github.com/Shopify/polaris-internal/issues/1517

Open to other ideas on how to fix this but character width is the most
performant I can think of


https://github.com/Shopify/polaris/assets/6844391/268bc7c8-48ef-43ba-895f-b647a5a7b8c1
